### PR TITLE
Add ClickHouse support

### PR DESCRIPTION
**Pull request by tolkonepiu**
Add support for [ClickHouse](https://clickhouse.yandex/)

### CODE REVIEW

File: modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java [hunk 1]
```diff
@@ -78,9 +81,9 @@ public void test() throws SQLException {
         }
     }
 
-    private void performSimpleTest(String jdbcUrl) throws SQLException {
+    private void performSimpleTest(String jdbcUrl, boolean pmdKnownBroken) throws SQLException {
```
Comment (kiview) [inline modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java]: I'm not so happy with adding an additional param to the method if it's only needed by on tested class.
The tests will fail if not set?

File: modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Override
+    protected Integer getLivenessCheckPort() {
+        return getMappedPort(HTTP_PORT);
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return "ru.yandex.clickhouse.ClickHouseDriver";
```
Comment (kiview) [inline modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java]: Please put the String into a constant.

File: modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public static final String IMAGE = "yandex/clickhouse-server";
+    public static final Integer HTTP_PORT = 8123;
+    public static final Integer NATIVE_PORT = 9000;
+
+    private String databaseName = "default";
+    private String username = "default";
+    private String password = "";
+
+    public ClickHouseContainer() {
+        super(IMAGE + ":1.1.54310");
```
Comment (kiview) [inline modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java]: Default Tag in constant please.

File: modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return "ru.yandex.clickhouse.ClickHouseDriver";
+    }
+
+    @Override
+    public String getJdbcUrl() {
+        return "jdbc:clickhouse://" + getContainerIpAddress() + ":" + getMappedPort(HTTP_PORT) + "/" + databaseName;
```
Comment (kiview) [inline modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java]: JDBC url prefix in constant.

File: modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }
+
+    @Override
+    public String getTestQueryString() {
+        return "SELECT 1";
```
Comment (kiview) [inline modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java]: In order to be consistent, the query String could go into a constant as well :slightly_smiling_face:

**Comment by rnorth** [comment]
We're shortly going to be merging #574, which changes our build system to Gradle. This is in part intended to make contributions of modules easier (per #564), but unfortunately means that for a short while your PR is going to show merge conflicts with the master branch.

I just want to let you know we don't want to create new work for you: *we'll take care of the merge conflicts shortly*. Please don't worry - we're grateful for your PR and want to help integrate it soon. Thank you.

**Comment by VladRassokhin** [comment]
@tolkonepiu Don't you mind if I'd fix code review problems and open new pull request base on your changes?

File: modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java (hunk 1 above)
Comment (tolkonepiu) [reply modules/jdbc-test/src/test/java/org/testcontainers/jdbc/JDBCDriverTest.java]: Method **getParameterMetaData** not implemented on clickhouse jdbc driver.

**Comment by mchernyakov** [comment]
@bsideup @VladRassokhin @tolkonepiu Hi guys. 
Is there any progress on this pull request?

**Comment by VladRassokhin** [comment]
I'll rebase code onto master, test it and submit separate PR by the end of this week

**Comment by bsideup** [comment]
superseded by #846

**Pull request closed without merging**